File: wf-api-integration/references/transfer/consult-transfer/golang/model/request/consult_transfer_request.go
package request

import (
	"fmt"

	"{moduleName}/wf/model/domain"
)

// ConsultTransferRequest represents the request for consultTransfer API.
// Used to inquire about transfer information (exchange rate, fees, etc.) before calling createTransfer.
type ConsultTransferRequest struct {
	TransferFromDetail *domain.TransferFromDetail `json:"transferFromDetail"`
	TransferToDetail   *domain.TransferToDetail   `json:"transferToDetail"`
}

// Validate validates the request parameters
func (r *ConsultTransferRequest) Validate() error {
	if r.TransferFromDetail == nil || r.TransferFromDetail.TransferFromAmount == nil ||
		r.TransferFromDetail.TransferFromAmount.Currency == "" {
		return fmt.Errorf("transferFromDetail.transferFromAmount.currency is required")
	}
	if r.TransferToDetail == nil || r.TransferToDetail.TransferToAmount == nil ||
		r.TransferToDetail.TransferToAmount.Currency == "" {
		return fmt.Errorf("transferToDetail.transferToAmount.currency is required")
	}
	// at least one value must be specified
	fromHasValue := r.TransferFromDetail.TransferFromAmount.Value != nil
	toHasValue := r.TransferToDetail.TransferToAmount.Value != nil
	if !fromHasValue && !toHasValue {
		return fmt.Errorf("either transferFromAmount.value or transferToAmount.value must be specified")
	}
	return nil
}

File: wf-api-integration/references/transfer/consult-transfer/java/model/request/ConsultTransferRequest.java
package {basePackage}.wf.model.request;

import {basePackage}.wf.model.domain.TransferFromDetail;
import {basePackage}.wf.model.domain.TransferToDetail;

import org.apache.commons.lang3.builder.ToStringBuilder;
import org.apache.commons.lang3.builder.ToStringStyle;

/**
 * WorldFirst consultTransfer 请求对象（转账咨询场景）
 *
 * <p>在调用 createTransfer API 进行转账之前，集成商可调用此接口获取转账相关信息，
 * 如跨币种汇率、手续费等。
 *
 */
public class ConsultTransferRequest {

    /** 支付方转账详情 */
    private TransferFromDetail transferFromDetail;

    /** 收款方转账详情 */
    private TransferToDetail transferToDetail;

    public TransferFromDetail getTransferFromDetail() {
        return transferFromDetail;
    }

    public void setTransferFromDetail(TransferFromDetail transferFromDetail) {
        this.transferFromDetail = transferFromDetail;
    }

    public TransferToDetail getTransferToDetail() {
        return transferToDetail;
    }

    public void setTransferToDetail(TransferToDetail transferToDetail) {
        this.transferToDetail = transferToDetail;
    }

    @Override
    public String toString() {
        return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
    }
}
